campaignion_tracking/campaignion_tracking.module

```diff
@@ -126,9 +126,13 @@ function campaignion_tracking_webform_redirect_alter($redirect, Submission $subm
   $webformFragments[] = 'sid=' . $submission->sid;
   $webformFragments[] = 'title=' . rawurlencode($submission->node->title);
 
+  $channels = campaignion_opt_in_channel_info();
   // Add optin values for all defined optin fields
-  foreach ($submission->opt_in->values() as $optin) {
-    $webformFragments[] = 'optin[' . rawurlencode($optin['channel']) . ']=' . rawurlencode($optin['value']);
+  foreach ($channels as $channel => $title) {
```

Comment:
You could use `array_keys()` to avoid defining the unused variable `$title.`

Also you could loop over `array_keys(campaignionopt_in_channel_info())` unless you want to keep `$channels` for readability.

Reply:
thanks, keeping `$channels`